File: valid-anagram/ValidAnagram.c
bool isAnagram(char * s, char * t){
    int sLength = strlen(s);
    int tLength = strlen(t);

    if(sLength != tLength)
        return false;
    
    int sArray[26]={0}; // ascii values of 26 letters will be used as index
    int tArray[26]={0};

    for(int i=0; i < sLength; i++){
        sArray[*(s+i)- 'a']++;
        tArray[*(t+i)- 'a']++;
    }

    for(int i=0; i < 26; i++){
        if( sArray[i] != tArray[i])
            return false;
    }

    return true;
}

File: valid-anagram/ValidAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] sArray = new int[26];
        int[] tArray = new int[26];
        if(s.length() != t.length())
            return false;

        int size = s.length();

        for(int i=0; i < size; i++){
            sArray[s.charAt(i) - 'a']++;
            tArray[t.charAt(i) - 'a']++;    
        }

        for(int i = 0; i < 26; i++){
            if(sArray[i] !=  tArray[i] ){
                return false;
            }
        }
        return true;
    }
}